ui/ducks/send/send.js

```diff
@@ -2381,12 +2383,20 @@ export function signTransaction() {
         ),
       );
 
-      dispatch(
+      const { id } = await dispatch(
         addTransactionAndRouteToConfirmationPage(txParams, {
           sendFlowHistory: draftTransaction.history,
           type: transactionType,
         }),
       );
+
+      dispatch(
```

Comment:
I noticed we're not using await with dispatch here, but we do throughout the code above. Looking at the return type of signTransaction, dispatch isn't even async. Could this be a good chance to make things more consistent and less confusing in this code?

Reply:
Sure, added `await` to `dispatch(setMaxValueMode(/* ... */)))`

Reply:
Is this not just a side effect of the action you are dispatching?

Some actions will simply update state entirely synchronously and so don't need an `await` where as many will submit a request to the background and therefore return a promise that can be awaited if needed.

Reply:
Maybe I misunderstood Derek here, for sure `await` will not have any effect at all.